icrestimator.py

```diff
@@ -335,15 +335,10 @@ def flip_wheel(self, q: np.ndarray, S_lmda: np.ndarray):
         We also track the number of times each wheel has been flipped to ensure that it drives
         in the correct direction, True indicates drive direction should be reversed.
         """
-        output = np.array([None] * self.n_modules)
-        for module in range(self.n_modules):
-            dif = q[module] - S_lmda[module]
-            output[module] = math.atan(math.sin(dif) / math.cos(dif))
-            absolute_direction = math.atan2(math.sin(dif), math.cos(dif))
-            if abs(output[module] - absolute_direction) > self.tolerance:
-                self.flipped[module] = True
-            else:
-                self.flipped[module] = False
+        dif = q - S_lmda
+        output = np.arctan(np.sin(dif) / np.cos(dif))
+        absolute_direction = np.arctan2(np.sin(dif), np.cos(dif))
```

Comment:
Alright, cool. You might notice now that all the sines and cosines here are being computed twice: that's probably once too many.

Reply:
Are parameters expensive in terms of computation? Because we only use the difference between q and S(lmda) we could take that as a single input as opposed to two arrays, doing this would probably reduce readability though.